model/slot.rb

```diff
@@ -8,4 +9,21 @@ def validate
     super
     validates_unique :car_id
   end
+
+  def self.get_slot(registration_number)
+    where(
+      car_id: DATABASE[:cars].select(:id).where(registration_number:)
+    ).call(:first)
+  end
+
+  def self.unpark_car(slot)
+    invoice_id = Invoice.create(slot[:car_id], slot[:entry_time]).id
+    invoice = Invoice.where(id: invoice_id).join(
```

Comment:
You can use Invoice.find_by_id method. You already have the id. The logic is duplicated here.

Reply:
I am actually using the sequel gem instead of ActiveRecord..

Reply:
You have the same logic in Invoice class method `find_by_id(invoice_id)` - Why are you not able to use the method in the model? Why are you repeating the logic here?